Add keycode in options

nckenn opened the pull request:
trigger only search when keycode specified, for example { keyCode : 13 } for enter  event

nckenn committed:
Add keycode in options

trigger only search when keycode specified, for example { keyCode : 13 } for enter  event

nitriques reviewed (changes requested):
Thanks! That's cool! But, before I merge, can you update your PR (by adding a new commit) that properly formats the code ? You are missing a bunch of spaces between statements and () {} . Thanks!

nckenn committed:
Add keycode in options

trigger only search when keycode specified.
for example using enter event :

$('input#search').quicksearch('table tbody tr', {
	'delay': 100,
	'selector': 'th',
	'stripeRows': ['odd', 'even'],
	'loader': 'span.loading',
	'noResults': 'tr#noresults',
	'bind': 'keyup keydown',
	'minValLength': 2,
	'removeDiacritics': true,
        'keyCode ': 13,
	'onBefore': function () {
		console.log('on before');
	},
	'onAfter': function () {
		console.log('on after');
	},
	'onValTooSmall': function (val) {
		console.log('value ' + val + ' is too small');
	},
	'show': function () {
		$(this).addClass('show');
	},
	'hide': function () {
		$(this).removeClass('show');
	}
	'prepareQuery': function (val) {
		return new RegExp(val, "i");
	},
	'testQuery': function (query, txt, _row) {
		return query.test(txt);
	}
});

nckenn commented:
Hi, hope its ok now :D

nitriques commented on src/jquery.quicksearch.js:
```diff
@@ -316,9 +317,18 @@
-				val = $(this).val();
-				self.trigger();
+				if (options.keyCode) {
+				    var keycode = (e.keyCode ? e.keyCode : e.which);
```
This could be written `e.keyCode || e.which` :P

nitriques commented:
Thanks Yoh !

nitriques commented:
Version 2.4.0 is on npm!
